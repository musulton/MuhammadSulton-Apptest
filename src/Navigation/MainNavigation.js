import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

import ScreenConfigs from './ScreenConfigs';

const Stack = createNativeStackNavigator();

const MainNavigation = () => (
  <NavigationContainer>
    <Stack.Navigator>
      {ScreenConfigs.map(item => (
        <Stack.Screen
          name={item.name}
          component={item.component}
          key={item.name}
        />
      ))}
    </Stack.Navigator>
  </NavigationContainer>
);

export default MainNavigation;
