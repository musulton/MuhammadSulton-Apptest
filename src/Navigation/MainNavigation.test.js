import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import MainNavigation from './MainNavigation';

jest.mock('../Containers', () => ({
  ContactList: 'ContactList',
  ContactForm: 'ContactForm',
  ContactDetails: 'ContactDetails',
}));

describe('MainNavigation', () => {
  afterEach(cleanup);

  it('should render MainNavigation correctly', () => {
    const {toJSON} = render(<MainNavigation />);

    expect(toJSON()).toMatchSnapshot();
  });
});
