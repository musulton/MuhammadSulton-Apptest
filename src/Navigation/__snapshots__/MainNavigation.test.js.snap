// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation should render MainNavigation correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="List"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <ContactList
          navigation={
            Object {
              "addListener": [Function],
              "canGoBack": [Function],
              "dispatch": [Function],
              "getParent": [Function],
              "getState": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "navigate": [Function],
              "pop": [Function],
              "popToTop": [Function],
              "push": [Function],
              "removeListener": [Function],
              "replace": [Function],
              "reset": [Function],
              "setOptions": [Function],
              "setParams": [Function],
            }
          }
          route={
            Object {
              "key": "List-TbrK_4amHHo3zybfcU2dq",
              "name": "List",
              "params": undefined,
            }
          }
        />
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
